lib/modules/datasource/packagist/index.ts

```diff
@@ -118,39 +118,37 @@ export class PackagistDatasource extends Datasource {
       packages,
       providersUrl,
       providersLazyUrl,
-      files,
-      includesFiles,
-      providerPackages,
+      providerIncludes,
+      includes,
+      providers,
     } = registryMeta;
-    if (files) {
-      const queue = files.map(
-        (file) => (): Promise<PackagistFile> =>
-          this.getPackagistFile(regUrl, file)
-      );
-      const resolvedFiles = await p.all(queue);
-      for (const res of resolvedFiles) {
-        for (const [name, val] of Object.entries(res.providers)) {
-          providerPackages[name] = val.sha256;
-        }
-      }
-    }
+
     const includesPackages: Record<string, ReleaseResult> = {};
-    if (includesFiles) {
-      for (const file of includesFiles) {
-        const res = await this.getPackagistFile(regUrl, file);
-        if (res.packages) {
-          for (const [key, val] of Object.entries(res.packages)) {
-            const dep = PackagistDatasource.extractDepReleases(val);
-            includesPackages[key] = dep;
-          }
+
+    const providerIncludesTasks = providerIncludes.map((file) => async () => {
```

Comment:
should we limit concurrency via `pMap` 🤔

Reply:
Default value of `5` looks good, I think

Reply:
i think you are wrong. this `Array.map` is executing the callback directly. so you get an array for already executing promises. so the `p.all` below just waits for all promises to resolve / reject. so we've no concurrency limit at all.

Reply:
🙈  i now see the double `=>`.  code is hard to read. can we extract the callbacks to separate functions to make it more clear?